ark/utils/test_utils.py

```diff
@@ -580,6 +581,52 @@ def create_test_extraction_data():
     return cell_mask, channel_data
 
 
+def save_figure(save_dir, save_file):
+    """Verify save_dir and save_file, then save to specified location
+
+    Args:
+        save_dir (str):
+            the name of the directory we wish to save to
+        save_file (str):
+            the name of the file we wish to save to
+    """
+
+    # verify save_dir exists
+    if not os.path.exists(save_dir):
+        raise FileNotFoundError("save_dir %s does not exist" % save_dir)
+
+    # verify that if save_dir specified, save_file must also be specified
+    if save_file is None:
+        raise FileNotFoundError("save_dir specified but no save_file specified")
+
+    plt.savefig(os.path.join(save_dir, save_file))
+
+
+def verify_in_list(list_verify, values_bucket, name_list_verify, name_values_bucket):
+    """Verify at least whether the values in list_verify exist in values_bucket
+
+    Args:
+        list_verify (list):
+            the list we want to check for all values existing in values_bucket
+        values_bucket (list):
+            the list of acceptable values that should only be appearing in list_verify
+        name_list_verify (str):
+            should be similar to the name of the variable associated with list_verify
+        name_values_bucket (str):
+            should be similar to the name of the variable associated with values_bucket
+
+    Raises:
+        ValueError:
+            if not all values in list_verify exist in values_bucket
+    """
+
+    if not np.isin(list_verify, values_bucket).all():
```

Comment:
Nice, this is much closer to what I had in mind. This will be a super useful, general function to give more descriptive error messages. 

Right now, it only supports checking for subsets of list 2 being present in list 1. Another use case would be adding checks for lists that should have all of the same items. 
For example, here: https://github.com/angelolab/ark-analysis/blob/526ea6d2e7d6446b618e52cc8fabed302b24fd9f/ark/segmentation/marker_quantification.py#L167

I think adding a `mode` flag that controls what type of check is being performed would accomplish this.

Reply:
My concern with this approach is that `verify_in_list` can get very confusing the more types of checks we add. In this case, for example, we're not really verifying whether something exists in another list, rather, it's more like if both lists exist inside each other. Even if we choose a different name for the function and the arg names to make things clearer (which I will be addressing), it can be really easy to forget which mode to set for this function, especially if we need to add more to it.

I do like the idea of having this check, but I think it's worth looking into making this into its own separate function if we expect set equality to pop up more often (same goes for other list-based verifications that pop up). I actually think checking if a value is a proper index into a list of keys (ex. val in dict.keys(), xarray.coords, DataFrame.columns or DataFrame.index) should also be its own separate function (we do that plenty throughout the repo). This would have the benefit of not only making it clear which function to call, but also prevent having too many mixed type arguments and allow for the customization of error messages without needing a ton of control statements.

It's probably not a big deal if we do choose to put everything into the same function, I just think it might make life a bit easier for us and for later contributors if we split things up a bit.

Reply:
Totally agree that checking if two lists are sets with the same items is a different functionality. Lets put that in a differently named function to make our life easier. 

I'm not sure what the difference is between the function that you currently have written and one that would check if an item is a valid index into a list of keys; isn't that just checking if it's present in the list?

Reply:
@ngreenwald I can keep the valid index checking in `verify_in_list`.